### PR TITLE
python3Packages.lime: init at 0.2.0.1

### PR DESCRIPTION
**Pull request by khaser**
###### Description of changes

[Lime](https://github.com/marcotcr/lime): python module for explaining the predictions of any machine learning classifier 
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes (or backporting 23.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/lime/default.nix [hunk 1]
```diff
@@ -0,0 +1,48 @@
+{ lib
+, fetchPypi
+, buildPythonPackage
+
+, matplotlib, numpy, scipy, tqdm, scikit-learn-extra, scikit-image
```
Comment (natsukium) [inline pkgs/development/python-modules/lime/default.nix:5]: One item per line is preferred for future diff.

File: pkgs/development/python-modules/lime/default.nix [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    inherit pname version;
+    hash = "sha256-dpYOTwVf61Pom1AiODuvyHtj8lusYmWYSwozPRpX94E=";
+  };
+
+  propagatedBuildInputs =  [
+    matplotlib
+    numpy
+    scipy
+    tqdm
+    scikit-learn-extra
```
Comment (natsukium) [inline pkgs/development/python-modules/lime/default.nix:28]: Do we need `scikit-learn-extra`?
It seems that lime itself depends on just `scikit-learn`.

File: pkgs/development/python-modules/lime/default.nix [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+  nativeCheckInputs = [
+    pytestCheckHook
+  ];
+
+  disabledTestPaths = [
+    # touches network
+    "lime/tests/test_lime_text.py"
+  ];
+
```
Comment (natsukium) [inline pkgs/development/python-modules/lime/default.nix:40]: Would you please add `pythonImportsCheck`?

**Review by natsukium — COMMENTED** [review]
I would like this library in nixpkgs, too.
Thank you.

File: pkgs/development/python-modules/lime/default.nix (hunk 3 above)
Comment (khaser) [reply pkgs/development/python-modules/lime/default.nix:45]: Of course, I have added modules listed in [module doc's](https://github.com/marcotcr/lime/blob/master/doc/lime.rst) to `pythonImportsCheck`. Thanks for fast review.

File: pkgs/development/python-modules/lime/default.nix (hunk 2 above)
Comment (khaser) [reply pkgs/development/python-modules/lime/default.nix]: You are right, we don't need `scikit-learn-extra`. Fixed

**Comment by natsukium** [comment]
@ofborg build python310Packages.lime
@ofborg build python311Packages.lime

**Comment by natsukium** [comment]
Result of `nixpkgs-review pr 237973` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python310Packages.lime</li>
    <li>python310Packages.lime.dist</li>
    <li>python311Packages.lime</li>
    <li>python311Packages.lime.dist</li>
  </ul>
</details>

**Review by natsukium — APPROVED** [review]
LGTM, thanks.

**Pull request merged**